integration-tests/bats/remotes-push-pull.bats

```diff
@@ -0,0 +1,845 @@
+    run dolt remote -v
+    [[ "$output" =~  origin ]] || false
+    run dolt push origin
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "fatal: The current branch main has no upstream branch." ]] || false
+}
+
+@test "remotes-push-pull: push not specifying a branch create new remote branch without setting upstream on non-default remote" {
+    dolt remote add origin http://localhost:50051/test-org/test-repo
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    run dolt remote -v
+    [[ "$output" =~  origin ]] || false
+    [[ "$output" =~  test-remote ]] || false
+
+    dolt checkout -b test-branch
+    dolt branch -a
+    run dolt push test-remote
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "* [new branch]          test-branch -> test-branch" ]] || false
+}
+
+@test "remotes-push-pull: push not specifying a branch creates new remote branch and sets upstream on non-default remote when -u is used" {
+    dolt remote add origin http://localhost:50051/test-org/test-repo
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    run dolt remote -v
+    [[ "$output" =~  origin ]] || false
+    [[ "$output" =~  test-remote ]] || false
+
+    dolt checkout -b test-branch
+    run dolt push -u test-remote
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "* [new branch]          test-branch -> test-branch" ]] || false
+    [[ "$output" =~ "branch 'test-branch' set up to track 'test-remote/test-branch'." ]] || false
+}
+
+@test "remotes-push-pull: push origin throws an error when no remote is set" {
+    run dolt remote -v
+    [ "$output" = "" ]
+    run dolt push origin
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "fatal: 'origin' does not appear to be a dolt repository" ]] || false
+}
+
+@test "remotes-push-pull: push --all with no remote or refspec specified" {
+    dolt remote add origin http://localhost:50051/test-org/test-repo
+    run dolt remote -v
+    [[ "$output" =~  origin ]] || false
+
+    dolt checkout -b new-branch
+    run dolt push --all
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "* [new branch]          main -> main" ]] || false
+    [[ "$output" =~ "* [new branch]          new-branch -> new-branch" ]] || false
+}
+
+@test "remotes-push-pull: push --all with remote specified" {
+    mkdir remote
+    mkdir repo1
+
+    cd repo1
+    dolt init
+    dolt remote add origin file://../remote
+    dolt push origin main
+
+    cd ..
+    dolt clone file://./remote repo2
+    cd repo2
+
+    dolt sql -q "CREATE TABLE test (pk INT PRIMARY KEY, col1 VARCHAR(10))"
+    dolt add .
+    dolt commit -am "create table"
+    dolt checkout -b branch1
+    dolt sql -q "INSERT INTO test VALUES (1, '1')"
+    dolt commit -am "add 1s"
+    run dolt push --all origin  # should not set upstream for new branches
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ " * [new branch]          branch1 -> branch1" ]] || false
+    [[ ! "$output" =~ "branch 'branch1' set up to track 'origin/branch1'." ]] || false
```

Comment:
Do we have any other ways to see the default upstream tracking branch? Does that show up in `dolt config --list --local`? This negative string matching is just a little fragile, since only this exact string will fail the test, so if the message changes slightly, we wouldn't catch it that a default upstream had been set. Being able to check that more directly would help make this test a little sturdier.

Reply:
Currently, the branch config is not shown with `dolt config --list`. I think is only shows ones in `Valid configuration variables`, which can be found with `dolt config -h`.
I updated it to more specific check for upstream set for this branch.